lib/api/core/entrypoints/embedded/protocols/http.js

```diff
@@ -374,7 +382,7 @@ class HttpProtocol extends Protocol {
       };
 
     if (typeof allowCompression !== 'boolean') {
-      throw new Error('Invalid HTTP "allowCompression" parameter value: expected a boolean value');
+      throw new KuzzleInternalError('Invalid HTTP "allowCompression" parameter value: expected a boolean value');
```

Comment:
No, this is not a Kuzzle internal error, but a configuration error (which is probably a lacking Error object).

This doesn't really matter because, in the end, the error is caught during initialization, and that aborts Kuzzle's start sequence. The error type is only really useful when a response is sent back to a client, or when it's a plugin error.

Reply:
Actually I opted for a different approach myself, while working on you-know-what: I'm using the `assert` native module for checks made during startup.
It's cleaner as in most cases we expect failed assertions to stop the program execution at some point.

Reply:
I agree with the usage of `assert`